src/main/java/org/openrewrite/java/migrate/lang/UseVarKeyword.java

```diff
@@ -0,0 +1,162 @@
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.migrate.lang;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.jetbrains.annotations.NotNull;
+import org.openrewrite.Cursor;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.JavaVisitor;
+import org.openrewrite.java.search.HasJavaVersion;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaType;
+import org.openrewrite.java.tree.TypeTree;
+
+import java.time.Duration;
+
+import static java.lang.String.format;
+import static java.util.Objects.*;
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class UseVarKeyword extends Recipe {
+    public String getDisplayName() {
+        return "Use local variable type-inference (var) where possible";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Local variable type-inference reduce the noise produces by repeating the type definitions in Java 10 or higher.";
+    }
+
+    @Override
+    public @Nullable Duration getEstimatedEffortPerOccurrence() {
+        return Duration.ofMinutes(1);
+    }
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getSingleSourceApplicableTest() {
+        return new HasJavaVersion("10", true).getVisitor();
+    }
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getVisitor() {
+        // J.VariableDeclarations
+        return new JavaVisitor<ExecutionContext>() {
+            private final JavaType.Primitive SHORT_TYPE = JavaType.Primitive.Short;
+            private final JavaType.Primitive BYTE_TYPE = JavaType.Primitive.Byte;
+            private final JavaTemplate template = JavaTemplate.builder(this::getCursor, "var #{} = #{any()}")
+                    .javaParser(JavaParser.fromJavaVersion()).build();
+
+            @Override
+            public J.VariableDeclarations visitVariableDeclarations(J.VariableDeclarations multiVariable, ExecutionContext executionContext) {
+                J.VariableDeclarations vd = (J.VariableDeclarations) super.visitVariableDeclarations(multiVariable, executionContext);
+
+                boolean isOutsideMethode = !determineIfIsInsideMethode(this.getCursor());
```

Comment:
In a few places we can replace `methode` with `method` (also `Methode` with `Methode`).

Reply:
🫣